src/conn/conn_api.c

```diff
@@ -1310,6 +1310,8 @@ __conn_open_session(WT_CONNECTION *wt_conn, WT_EVENT_HANDLER *event_handler, con
 
     session_ret = NULL;
     WT_ERR(__wt_open_session(conn, event_handler, config, true, &session_ret));
+    //set the open_session method's session name
+    session_ret->name = "connection-open-session";
```

Comment:
@y123456yz I think the intention of the change is to distinguish internal sessions from user-created sessions. 

If I have understood your intention correctly, we already have a flag for this distinction - `WT_SESSION_INTERNAL`

Reply:
@Siddhartha8899
thanks for you reply, WT_SESSION_INTERNAL can distinction if the session is internal or user created session, but many code shoud print the session's name .

for example: in __wt_verbose_dump_sessions func, 
        WT_ERR(__wt_msg(session, "  Name: %s", s->name == NULL ? "EMPTY" : s->name));

the user session name will be "EMPTY", It doesn't feel very friendly for help us to analyze.

Reply:
Thank you for clarifying the intention – I now understand. However, considering that this change (setting name to NULL) dates back almost a decade, I would discourage making a change like this as modifying this would require a full review of all instances where we check for `session->name`, as there might be dependencies on the NULL value. Altering this could potentially introduce errors.

@agorrod thoughts?

Reply:
@Siddhartha8899
@agorrod
I wonder if we can use another way, for example, we add following code:
`const char*
__wt_get_session_name(WT_SESSION_IMPL *session)
{
    if (session->name)
        return session->name;

    if (!F_ISSET(session, WT_SESSION_INTERNAL)) {
        return "user-open-session"
    }

    return "EMPTY"
}
`

if we shoud print the session name, wu use this function?  This can avoids potential undiscovered problems.

can you give some suggestions?

Reply:
I see all the code that use the sesion‘s name,  it should be safe。

Reply:
@agorrod 
can you review this PR, thanks.

Reply:
@y123456yz thanks for your patience. We will get to this by the end of next week.

Reply:
@Siddhartha8899 
If it's convenient for you, I wonder if you could help review the following PR:

https://github.com/wiredtiger/wiredtiger/pull/9653
https://github.com/wiredtiger/wiredtiger/pull/9654
https://github.com/wiredtiger/wiredtiger/pull/9658